src/test/java/bio/terra/service/iam/sam/SamIamTest.java

```diff
@@ -57,4 +135,274 @@ public void testExtractErrorMessageDeepNested() {
+
+        final UUID profileId = UUID.randomUUID();
+        samIam.deleteProfileResource(userReq, profileId.toString());
+        verify(samResourceApi, times(1))
+            .deleteResource(eq(IamResourceType.SPEND_PROFILE.getSamResourceName()), eq(profileId.toString()));
+    }
+
+    @Test
+    public void testRetrievePoliciesAndEmails() throws ApiException, InterruptedException {
+        final UUID id = UUID.randomUUID();
+        final String policyEmail = "policygroup@firecloud.org";
+        final String memberEmail = "a@a.com";
+        when(samResourceApi.listResourcePolicies(
+            IamResourceType.SPEND_PROFILE.getSamResourceName(),
+            id.toString()
+        )).thenReturn(Collections.singletonList(
+            new AccessPolicyResponseEntry()
+                .policyName(IamRole.CUSTODIAN.toString())
+                .email(policyEmail)
+                .policy(new AccessPolicyMembership().addMemberEmailsItem(
+                    memberEmail
+                ))
+        ));
+
+        assertThat(samIam.retrievePolicies(userReq, IamResourceType.SPEND_PROFILE, id))
+            .isEqualTo(Collections.singletonList(
+                new PolicyModel()
+                    .name(IamRole.CUSTODIAN.toString())
+                    .addMembersItem(memberEmail)
+            ));
+
+        assertThat(samIam.retrievePolicyEmails(userReq, IamResourceType.SPEND_PROFILE, id))
+            .isEqualTo(Maps.newHashMap(IamRole.CUSTODIAN, policyEmail));
+    }
+
+    @Test
+    public void testCreateDataset() throws InterruptedException, ApiException {
+        final UUID datasetId = UUID.randomUUID();
+        // Note: in our case, policies have a 1:1 relationship with roles
+        final List<IamRole> allPolicies = Arrays.asList(
+            IamRole.ADMIN,
+            IamRole.STEWARD,
+            IamRole.CUSTODIAN,
+            IamRole.SNAPSHOT_CREATOR
+        );
+        final List<IamRole> syncedPolicies = Arrays.asList(
+            IamRole.STEWARD,
+            IamRole.CUSTODIAN,
+            IamRole.SNAPSHOT_CREATOR
+        );
+
+        for (IamRole policy : allPolicies) {
+            when(samGoogleApi.syncPolicy(
+                IamResourceType.DATASET.getSamResourceName(),
+                datasetId.toString(),
+                policy.toString())
+            ).thenReturn(Maps.newHashMap("policygroup-" + policy + "@firecloud.org", Collections.emptyList()));
+        }
+
+        assertThat(samIam.createDatasetResource(userReq, datasetId))
+            .isEqualTo(syncedPolicies.stream().collect(Collectors.toMap(
+                p -> p,
+                p -> "policygroup-" + p + "@firecloud.org"
+            )));
+    }
+
+    @Test
+    public void testCreateSnapshot() throws InterruptedException, ApiException {
+        final UUID snapshotId = UUID.randomUUID();
+        // Note: in our case, policies have a 1:1 relationship with roles
+        final List<IamRole> allPolicies = Arrays.asList(
```

Comment:
a bunch of cases where `List.of()` can be used
```suggestion
        final List<IamRole> allPolicies = List.of(
```